CDCgov/prime-simplereport
# DB Anonymization

# DATABASE PULL REQUEST

## Related Issue

- Resolves #3878 
- We want to anonymize databases for testing against realistic databases.

## Changes Proposed

- Add a way to generate data to load into our anonymized db (not used yet)
- Add a way to anonymize a PostgreSQL db dump
- Add masking rules for all potentially sensitive columns and json blobs in our db

## Additional Information

- The data generation script is a WIP and is not in use, although it has been practically tested.
- Would love to figure out how to test our masking rules in an automated way and enforce having a masking rule for each column
- Should write tests for our data generation script but that was left out as it may be stripped or drastically changed.
- #3879 and #3289 are upcoming and will expand on this. Extensive testing will take place in lower environments.

## Testing

- Run the yarn commands while your local docker containers are running
- **Check the masking rules.**

## Checklist for Primary Reviewer

- [ ] Only database changes are included in this PR
- [ ] Any new tables or columns that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
- [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views (including re-granting permission to the no-PHI user if need be)
- [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] Rollback has been verifed locally and in a deployed environment
- [ ] Masking rules have been verified
----
